File: boj/11652.cpp
#include <iostream>
#include <algorithm>

using namespace std;

long long arr[100003];

int main(void) {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    long long n; cin >> n;
    for (int i = 0; i < n; ++i) cin >> arr[i];
    sort(arr, arr + n);
    long long ans = arr[0], maxCount = 1, temp = 1;
    for (int i = 1; i < n; ++i) {
        if (arr[i] == arr[i - 1]) ++temp;
        else temp = 1;
        if (maxCount < temp) {
            maxCount = temp;
            ans = arr[i];
        }
    }
    cout << ans << "\n";
    return 0;
}

File: boj/11652.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        long[] arr = new long[n];
        for(int i = 0; i < n; ++i) arr[i] = Long.parseLong(br.readLine());
        Arrays.sort(arr);
        long ans = arr[0], maxCount = 1, temp = 1;
        for(int i = 1; i < n; ++i) {
            if(arr[i] == arr[i - 1]) ++temp;
            else temp = 1;
            if(maxCount < temp) {
                maxCount = temp;
                ans = arr[i];
            }
        }
        System.out.println(ans);
    }
}
